numpy/core/src/multiarray/textreading/tokenize.cpp

```diff
@@ -62,8 +54,8 @@ copy_to_field_buffer_@type@(tokenizer_state *ts,
                     "line too long to handle while reading file.");
             return -1;
         }
-        Py_UCS4 *grown = PyMem_Realloc(ts->field_buffer, alloc_size);
-        if (grown == NULL) {
+        auto *grown = (Py_UCS4 *)PyMem_Realloc(ts->field_buffer, alloc_size);
```

Comment:
Do we have a decision about using features like this in C++ code? Personally I prefer `Py_UCS4`.